src/hooks/useQueryParam/index.ts

```diff
@@ -36,15 +44,39 @@ const useQueryParam = <T>(
     [setSearchParams, searchParams, param]
   );
 
-  return [
+  const queryParam =
     searchParams[param] !== undefined
       ? (conditionalToArray(
           searchParams[param],
           Array.isArray(defaultParam)
         ) as unknown as T)
-      : defaultParam,
-    setQueryParam,
-  ] as const;
+      : defaultParam;
+
+  return [queryParam, setQueryParam] as const;
+};
+
+/**
+ * `useFilterParam` is a specialized form of useQueryParam. It needs to do special processing when converting
+ * filters to and from URLs.
+ */
+const useFilterParam = () => {
+  const [searchParams, setSearchParams] = useQueryParams();
+
+  const parsedFilters = parseFilters(
+    conditionalToArray(searchParams.filters ?? [], true) as string[]
+  );
+
+  const setFiltersParam = useCallback(
+    (filters: Filters) => {
+      setSearchParams({
+        ...searchParams,
+        filters: stringifyFilters(filters),
+      });
+    },
+    [setSearchParams, searchParams]
+  );
+
+  return [parsedFilters, setFiltersParam] as const;
 };
```

Comment:
I think this makes sense to move to its own folder/file. useQueryParm is generic this seems more geared to a specific feature